packages/signature-controller/src/SignatureController.ts

```diff
@@ -292,8 +306,31 @@ export class SignatureController extends BaseControllerV2<
         'eth_sign requires 32 byte message hash',
       );
     }
+    const messageId = await this.#messageManager.addUnapprovedMessage(
+      messageParams,
+      req,
+    );
+
+    const messageParamsWithId = {
+      ...messageParams,
+      metamaskId: messageId,
+    };
+
+    const signaturePromise =
+      this.#messageManager.createMessageListener(messageParamsWithId);
+
+    try {
+      await this.#requestApproval(messageParamsWithId, ApprovalType.EthSign);
+    } catch (error) {
+      this.#cancelAbstractMessage(this.#messageManager, messageId);
+      throw ethErrors.provider.userRejectedRequest(
+        'User rejected the request.',
+      );
+    }
+
+    await this.#signMessage(cloneDeep(messageParamsWithId));
```

Comment:
Rather than cloning after sending this in the message, could we instead clone inside `requestApproval` and send the clone in the message?

Would also benefit from a comment to clarify why we need the clone to begin with as sending it through the messaging system makes the object read only.

Reply:
Fixed